pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -181,39 +181,79 @@ func (b *BridgePodInterface) discoverPodNetworkInterface() error {
 	return nil
 }
 
+func (b *BridgePodInterface) delegateIPAddress() bool {
+	if b.iface.Bridge.DelegateIp != nil {
+		return *b.iface.Bridge.DelegateIp
+	}
+	return true
+}
+
 func (b *BridgePodInterface) preparePodNetworkInterfaces() error {
+	delegateIp := b.delegateIPAddress()
+
 	// Remove IP from POD interface
 	err := Handler.AddrDel(b.podNicLink, &b.vif.IP)
 
 	if err != nil {
-		log.Log.Reason(err).Errorf("failed to delete link for interface: %s", podInterface)
+		log.Log.Reason(err).Errorf("failed to delete address for interface: %s", podInterface)
 		return err
 	}
 
-	// Set interface link to down to change its MAC address
-	err = Handler.LinkSetDown(b.podNicLink)
-	if err != nil {
-		log.Log.Reason(err).Errorf("failed to bring link down for interface: %s", podInterface)
-		return err
+	if delegateIp {
+		// Set interface link to down to change its MAC address
+		err = Handler.LinkSetDown(b.podNicLink)
+		if err != nil {
+			log.Log.Reason(err).Errorf("failed to bring link down for interface: %s", podInterface)
+			return err
+		}
+
+		_, err = Handler.SetRandomMac(podInterface)
+		if err != nil {
+			return err
+		}
+
+		err = Handler.LinkSetUp(b.podNicLink)
+		if err != nil {
+			log.Log.Reason(err).Errorf("failed to bring link up for interface: %s", podInterface)
+			return err
+		}
 	}
 
-	_, err = Handler.SetRandomMac(podInterface)
-	if err != nil {
+	if err := b.createDefaultBridge(); err != nil {
 		return err
 	}
 
-	err = Handler.LinkSetUp(b.podNicLink)
-	if err != nil {
-		log.Log.Reason(err).Errorf("failed to bring link up for interface: %s", podInterface)
+	// Set up IP address on the bridge
+	var bridgeAddr string
+	if delegateIp {
+		// Set fake ip on a bridge
+		bridgeAddr = bridgeFakeIP
+	} else {
+		// Move allocated IP address to the bridge
+		bridgeAddr = b.vif.IP.IPNet.String()
+	}
+	if err := b.addIPAddress(bridgeAddr); err != nil {
 		return err
 	}
 
-	if err := b.createDefaultBridge(); err != nil {
-		return err
+	if delegateIp {
+		// Start DHCP server to lease the allocated IP address
+		b.startDHCPServer()
+	} else {
+		// Restore default gateway inside the pod
```

Comment:
Is it because we've moved the ip from the nic to the bridge?
I'm wondering what about the other routes or these shouldn't be affected?

Reply:
Yeah good comment, it's possible we need to restore other routes. I will need to check. Maybe some routes point to other nodes and I will miss them now.